tests/test_csr.py

```diff
@@ -241,3 +246,193 @@ def test_admin_save_existing_csr_should_not_renew(admin_client):
     # CSR and PK should not be regenerated
     assert signing_request.csr == original_csr
     assert signing_request.private_key == original_pk
+
+
+@pytest.mark.django_db
+def test_saving_valid_cert_does_create_cert_instance_via_post(
+    admin_client,
+    temp_private_root,
+):
+    assert Certificate.objects.count() == 0
+
+    csr = SigningRequest.objects.create(
+        common_name="test.example.com",
+        organization_name="Test Organization",
+        state_or_province_name="Test State",
+        country_name="NL",
+        email_address="email@valid.com",
+    )
+
+    private_key = load_pem_x509_private_key(csr.private_key.encode())
+    # Here I did not use the fixture mkcert to generate the certificate
+    # This is because the fixture mkcert did not generate a valid certificate
+    # The public keys of the private key and the certificate did not match
+    pub_cert = (
+        x509.CertificateBuilder()
+        .subject_name(
+            x509.Name(
+                [x509.NameAttribute(x509.NameOID.COMMON_NAME, "test.example.com")]
+            )
+        )
+        .issuer_name(
+            x509.Name(
+                [x509.NameAttribute(x509.NameOID.COMMON_NAME, "test.example.com")]
+            )
+        )
+        .public_key(private_key.public_key())
+        .serial_number(x509.random_serial_number())
+        .not_valid_before(datetime.datetime.now())
+        .not_valid_after(datetime.datetime.now() + datetime.timedelta(days=365))
+        .add_extension(
+            x509.SubjectAlternativeName([x509.DNSName("localhost")]),
+            critical=False,
+        )
```

Comment:
this doesn't really make sense for TLS clients - they don't have a (public) DNS name